KIPs/kip-dreaft_sdk_common_architecture.md

```diff
@@ -0,0 +1,1623 @@
+
+#### AbstractFeeDelegatedWithRatioTransaction
+
+`AbstractFeeDelegatedWithRatioTransaction` is an abstract class that abstracts [Partial Fee Delegation Transaction](https://docs.klaytn.com/klaytn/design/transactions/partial-fee-delegation) classes. `AbstractFeeDelegatedWithRatioTransaction` is implemented by extending `AbstractFeeDelegatedTransaction`. All Partial Fee Delegation Transaction classes are implemented by extending `AbstractFeeDelegatedWithRatioTransaction`.
+
+##### Variable description
+
+| Variable | Description |
+| ----------- | ----------- |
+| feeRatio: String | The ratio that constitutes the proportion of the transaction fee the fee payer will be burdened with. The valid range of this ratio is between 1 and 99. The ratio of 0, or 100 and above are not allowed. |
+
+##### Method description
+
+None
+
+#### LegacyTransaction
+
+`LegacyTransaction` represents a [legacy transaction](https://docs.klaytn.com/klaytn/design/transactions/basic#txtypelegacytransaction). This class is implemented by extending [AbstractTransaction](#abstracttransaction).
+
+##### Variable description
+
+| Variable | Description |
+| ----------- | ----------- |
+| to: String | The account address that will receive the transferred value or smart contact address if a legacy transaction execute smart contract. If a legacy transaction deploys a smart contract, to with the default value "0x". |
+| input: String | Data attached to the transaction, used for smart contract deployment/execution. |
+| value: String | The amount of KLAY in peb to be transferred. |
+
+##### Method description
+
+| Method | Description |
+| ----------- | ----------- |
+| decode(rlpEncoded: String): LegacyTransaction | Decodes an RLP-encoded LegacyTransaction string, a raw transaction, and returns a LegacyTransaction instance. It throws an exception if the decoding is failed. |
+| appendSignatures(sig: SignatureData): void | Appends signature to the transaction. LegacyTransaction can only have one signature. It throws an exception if multiple signatures are assigned to a `signatures`. |
+| appendSignatures(sig: List&#60;SignatureData&#62;): void | Appends signature to the transaction. LegacyTransaction can only have one signature. It throws an exception if multiple signatures are assigned to a `signatures`. |
+| getRLPEncoding(): String | Returns an RLP-encoded LegacyTransaction string. It throws an exception if the variables required for encoding are not defined. |
+| getRLPEncodingForSignature(): String | Returns an RLP-encoded transaction string for making the signature of the transaction sender. Since the method of obtaining RLP-encoding for signature of LegacyTransaction is different from other transaction types, getRLPEncodingForSignature should be overrided. It throws an exception if the variables required for encoding are not defined. |
```

Comment:
```suggestion
| getRLPEncodingForSignature(): String | Returns an RLP-encoded transaction string for making the signature of the sender. Since the method of obtaining the RLP-encoding string for signature of LegacyTransaction is different from other transaction types, getRLPEncodingForSignature should be overridden. It throws an exception if the variables required for encoding are not defined. |
```